packages/web3-eth/src/rpc_method_wrappers.ts

```diff
@@ -627,9 +669,12 @@ export async function signTransaction<ReturnFormat extends DataFormat>(
 		web3Context.requestManager,
 		formatTransaction(transaction, DEFAULT_RETURN_FORMAT),
 	);
+	const unformattedResponse = isString(response as HexStringBytes)
```

Comment:
This change is what's handling the `SignedTransacitonInfo`:

```typescript
export interface SignedTransactionInfo {
	raw: HexStringBytes;
	tx: TransactionSigned;
}
```

that Geth returns. Our library method is supposed to return the same `SignTransactionInfo` object, but in this current implementation, there is a difference between the object we return and Geth returns. Geth returns an object such as:

```typescript
{
          type: '0x0',
          nonce: '0x0',
          gasPrice: '0x3b9aca01',
          maxPriorityFeePerGas: null,
          maxFeePerGas: null,
          gas: '0x5208',
          value: '0x1',
          input: '0x',
          v: '0xa95',
          r: '0xdf1960615fd05d894b67c49e03b7bc7e3908ad00b44f25ae85c17a99f47e577a',
          s: '0x327aa2eb8269907dfcca27582e391a764e49c04a8ef79757a06ccf23ddd74987',
          to: '0x0000000000000000000000000000000000000000',
          hash: '0x21adb30975cc47c25b28eda40bc940da309990aafec0a831f5627f848e64091d'
}
``` 

While the library returns whatever the user provided to the method (so we'll always be missing properties such as `v`, `r`, `s`, `hash`)

Reply:
We could add these missing properties to the user's transaction object before returning when only the raw signed transaction is returned from the RPC client